integ-test/src/test/java/org/opensearch/sql/sql/NestedIT.java

```diff
@@ -366,4 +400,105 @@ public void test_nested_in_where_as_predicate_expression_with_relevance_query()
+        + TEST_INDEX_NESTED_TYPE;
+    JSONObject result = executeJdbcRequest(query);
+
+    assertEquals(6, result.getInt("total"));
+    verifySchema(result,
+        schema("nested(message.author)", null, "keyword"),
+        schema("nested(message.dayOfWeek)", null, "long"),
+        schema("nested(message.info)", null, "keyword"),
+        schema("nested(comment.data)", null, "keyword"));
+    verifyDataRows(result,
+        rows("e", 1, "a", "ab"),
+        rows("f", 2, "b", "aa"),
+        rows("g", 1, "c", "aa"),
+        rows("h", 4, "c", "ab"),
+        rows("i", 5, "a", "ab"),
+        rows("zz", 6, "zz", new JSONArray(List.of("aa", "bb"))));
+  }
+
+  @Test
+  public void nested_function_all_deep_nested_subfields() {
+    String query = "SELECT nested(message.author.address.*) FROM "
+        + TEST_INDEX_MULTI_NESTED_TYPE;
+    JSONObject result = executeJdbcRequest(query);
+
+    assertEquals(6, result.getInt("total"));
+    verifySchema(result,
+        schema("nested(message.author.address.number)", null, "integer"),
+        schema("nested(message.author.address.street)", null, "keyword"));
+    verifyDataRows(result,
+        rows(1, "bc"),
+        rows(2, "ab"),
+        rows(3, "sk"),
+        rows(4, "mb"),
+        rows(5, "on"),
+        rows(6, "qc"));
+  }
+
+  @Test
+  public void nested_function_all_subfields_for_two_nested_fields() {
+    String query = "SELECT nested(message.*), nested(comment.*) FROM "
+        + TEST_INDEX_NESTED_TYPE;
+    JSONObject result = executeJdbcRequest(query);
+
+    assertEquals(6, result.getInt("total"));
+    verifySchema(result,
+        schema("nested(message.author)", null, "keyword"),
+        schema("nested(message.dayOfWeek)", null, "long"),
+        schema("nested(message.info)", null, "keyword"),
+        schema("nested(comment.data)", null, "keyword"),
+        schema("nested(comment.likes)", null, "long"));
+    verifyDataRows(result,
+        rows("e", 1, "a", "ab", 3),
+        rows("f", 2, "b", "aa", 2),
+        rows("g", 1, "c", "aa", 3),
+        rows("h", 4, "c", "ab", 1),
+        rows("i", 5, "a", "ab", 1),
+        rows("zz", 6, "zz", new JSONArray(List.of("aa", "bb")), 10));
+  }
+
+  @Test
+  public void nested_function_all_subfields_and_non_nested_field() {
+    String query = "SELECT nested(message.*), myNum FROM " + TEST_INDEX_NESTED_TYPE;
+    JSONObject result = executeJdbcRequest(query);
+
+    assertEquals(6, result.getInt("total"));
+    verifySchema(result,
+        schema("nested(message.author)", null, "keyword"),
+        schema("nested(message.dayOfWeek)", null, "long"),
+        schema("nested(message.info)", null, "keyword"),
+        schema("myNum", null, "long"));
+    verifyDataRows(result,
+        rows("e", 1, "a", 1),
+        rows("f", 2, "b", 2),
+        rows("g", 1, "c", 3),
+        rows("h", 4, "c", 4),
+        rows("i", 5, "a", 4),
+        rows("zz", 6, "zz", new JSONArray(List.of(3, 4))));
+  }
```

Comment:
Can you add tests with `Invalid use of expression` error? https://github.com/Bit-Quill/opensearch-project-sql/pull/278#issuecomment-1599275236